### PR TITLE
GEODE-5707: CommitConflictException Test case.

### PR DESCRIPTION
**Pull request by vaijira**
Create 2 async tasks to force a CommitConflictException message.
After the exception the following tx should work.

### CODE REVIEW

File: cppcache/integration-test-2/CommitConflictExceptionTest.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    cache.getCacheTransactionManager()->begin();
+    begunTx1.set_value();
+    region->put("one", "one");
+    ASSERT_EQ(std::future_status::ready,
+              putTx2.get_future().wait_for(debug_safe(minutes(1))));
+    putTx1.set_value();
+    ASSERT_EQ(std::future_status::ready,
+              committedTx2.get_future().wait_for(debug_safe(minutes(1))));
+    try {
```
Comment (pivotal-jbarrett) [inline cppcache/integration-test-2/CommitConflictExceptionTest.cpp]: Would `EXPECT_THROW` not work in this case?

Comment (vaijira) [reply cppcache/integration-test-2/CommitConflictExceptionTest.cpp]: The problem with EXPECT_THROW or ASSERT_THROW is that i cannot assert that is a CommitConflictException only an apache::geode::client::Exception (at least i couldn't make it work with apache::geode::client::CommitConflictException) and to be sure that is a CommitConflictException i have to examine the content.

**Comment by echobravopapa** [comment]
@vaijira any news on this PR?  if not please close and re-open when its active again, Thanks!

**Comment by vaijira** [comment]
I added a new test and reproduce the issue.

**Comment by pdxcodemonkey** [comment]
Took a look this morning and discovered the new test was always passing here, because the test code didn't _actually_ stop the server.  You'll want to cherry-pick over commit abaf4061514399b89949d9dfd3792208aee86cc7 from https://github.com/pdxcodemonkey/geode-native.git or apply a similar change to make the test work.  Can confirm, with the change the test is 100% accurate, always fails without the TXCleaner line and always passes with it.  Nicely done!

Thanks,
Blake

**Pull request merged**